x/dex/keeper/utils/wasm.go

```diff
@@ -42,7 +42,11 @@ func sudo(sdkCtx sdk.Context, k *keeper.Keeper, contractAddress []byte, wasmMsg
 	// Note that the limit will effectively serve as a soft limit since it's
 	// possible for the actual computation to go above the specified limit, but
 	// the associated contract would be charged corresponding rent.
-	tmpCtx := sdkCtx.WithGasMeter(sdk.NewGasMeter(sdkCtx.GasMeter().Limit()))
+	gasLimit, err := k.GetContractGasLimit(sdkCtx, contractAddress)
```

Comment:
why are we passing contractAddress here (and above in `sudo` args as `[]byte` instead of `AccAddress` which is a defined type for `[]byte` IIRC?

Reply:
updated